CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/svg/parser/SVGParser.java

```diff
@@ -0,0 +1,207 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+package au.gov.asd.tac.constellation.plugins.importexport.svg.parser;
+
+import au.gov.asd.tac.constellation.plugins.importexport.svg.SVGObject;
+import au.gov.asd.tac.constellation.utilities.text.SeparatorConstants;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * The SVGParser facilitates the translation of SVGfiles into
+ * an in memory format for easy manipulation and translation. 
+ * In contrast to other Constellation import parsers, this import parser does not 
+ * assume data is in a tabular format and as such will not extend the InportFileParser
+ * 
+ * @author capricornunicorn123
+ */
+public class SVGParser {
+
+    private SVGParser() {
+        throw new IllegalStateException("Utility class");
+    }
+    /**
+     * 
+     * Takes an input stream and translates it to an SVG object. 
+     * Will only translate to an SVG object if the file data is valid.
+     * Does not support multi-line tags.
+     * @param inputStream
+     * @return
+     * @throws IOException 
+     */
+    public static SVGObject parse(InputStream inputStream) throws IOException {
+        SVGObject currentElement = null; 
+        final Collection<SVGObject> roots = new HashSet<>();
+        
+        try (final BufferedReader br = new BufferedReader(new InputStreamReader(inputStream))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                final String svgElement = SVGParser.isolateSVGElement(line);
+                final boolean openTag = SVGParser.isOpenTag(svgElement);
+                final boolean closeTag = SVGParser.isCloseTag(svgElement);
+                
+                // Create a new SVGObject with the current SVGObject as the parent 
+                if (openTag) {
+                        
+                    SVGObject newObject = new SVGObject(
+                            SVGParser.getElementType(svgElement), 
+                            currentElement, 
+                            SVGParser.getElementAttributes(svgElement)
+                    );
+                    currentElement = newObject;
+                }
+                // Move back up one level to the curren objects parent
+                if (closeTag) {
+                    currentElement = currentElement.getParent();
+                } 
+                // This parser curently requires all lines with be an SVG tag as it does not support multi line tags.
+                if (!openTag && !closeTag) {
+                    throw new UnsupportedOperationException(String.format("This line could not be interpreted: %s", svgElement));
+                }
+                
```

Comment:
Possible bug here. If the layout.svg file is parsed, this code attempts to add the first SVG element (the one with the xmlns attribute) twice. Although it won't be added in twice because the Set is used to store the elements maybe refactor it so the second attempt is not made. This would reduce the chance of bugs and make the code clearer for anyone in who works on it in the future.